### ISSUE TITLE
Add support for state encryption

### ISSUE THREAD

**Issue by mtprovenir** [gamunu/vscode-opentofu#75]
### Extension Version

v0.2.0

### Problem Statement

It seems that the latest version available of this extension is complaining about the unexpected field `encryption` when attempting to use state encryption. This is valid OpenTofu as noted [here](https://opentofu.org/docs/intro/whats-new/#state-encryption), and it also runs as expected.

![Screenshot from 2024-05-20 14-30-42](https://github.com/gamunu/vscode-opentofu/assets/168411899/16ba23ed-cc83-4c0d-82a6-a250e17bc4a9)

### Expected User Experience

It should not underline the field in red and recognise it as an acceptable field.

### References

- https://opentofu.org/docs/intro/whats-new/#state-encryption

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by mtprovenir** [comment]
I wasn't sure whether this should be classified as a bug or not as a new version may not have come out since the feature was added upstream, so went with feature request. Apologies if this is incorrect.

**Comment by gamunu** [comment]
@mtprovenir Yes, this a feature request for @gamunu/opentofu-ls. The language server doesn't recognize the encryption block.
